add my unit-test

ermaksmak opened the pull request.

ermaksmak committed:
add my unit-test

irisqul commented on tests.py:
```diff
@@ -21,4 +21,94 @@ def test_add_new_book_add_two_books(self):
         assert len(collector.get_books_rating()) == 2
```
Нужно исправить: AttributeError: 'BooksCollector' object has no attribute 'get_books_rating'

ermaksmak replied:
Закомментировал этот код из примера

irisqul commented on tests.py:
```diff
@@ -21,4 +21,94 @@ def test_add_new_book_add_two_books(self):
+    # чтобы тесты были независимыми в каждом из них создавай отдельный экземпляр класса BooksCollector()
+
+    def test_add_new_book_invalid_length(self):
+        collector = BooksCollector()
```
Отлично: в каждом тесте создается собственный экземпляр BooksCollector, но можно лучше - перенести создание экземпляра в фикстуры, а не повторять это предусловие в каждом тесте

ermaksmak replied:
Фикстуры добавлю позже уже самостоятельно

irisqul commented on tests.py:
```diff
@@ -21,4 +21,94 @@ def test_add_new_book_add_two_books(self):
+
+    def test_add_new_book_invalid_length(self):
+        collector = BooksCollector()
+        collector.add_new_book('')
```
Нужно исправить: тест неатомарный - разные граничные значения - это разные тесты. Тут хорошо подойдёт параметризация, чтобы тест запустился несколько раз для каждого ГЗ

ermaksmak replied:
Разбил тест на два теста

irisqul commented on tests.py:
```diff
@@ -21,4 +21,94 @@ def test_add_new_book_add_two_books(self):
+        collector.set_book_genre('Гарри Поттер', 'Мультфильмы')
+        collector.add_new_book('Пила')
+        collector.set_book_genre('Пила', 'Ужасы')
+        assert 'Гарри Поттер' in collector.get_books_for_children() and 'Пила' not in collector.get_books_for_children()
```
Нужно исправить: тест неатомарен

ermaksmak replied:
Исправил атомарность разбил на 2 теста

ermaksmak committed:
adding info in readme.md, adding parameterization шin test.py and creating 2 tesfrom test_add_new_book_invalid_lengt
adding info in readme.md, adding parameterization in test.py and creating 4 new test for test_add_new_book_invalid_lengt and test_get_books_for_children_list_is_empty

ermaksmak committed:
adding info in readme.md, adding parameterization шin test.py and creating 2 tesfrom test_add_new_book_invalid_lengt
adding info in readme.md, adding parameterization шin test.py and creating 2 tesfrom test_add_new_book_invalid_lengt

ermaksmak committed:
adding info in readme.md, adding parameterization in test.py and creating 2 test for invalid_lenght and 2 test for children_list

ermaksmak committed:
adding info in readme.md, adding parameterization in test.py and creating 2 test for invalid_lenght and 2 test for children_list

ermaksmak committed:
Update README.md
